go.mod

```diff
@@ -1,8 +1,6 @@
 module github.com/crossplane/crossplane
 
-go 1.21
```

Comment:
[`lint` check](https://github.com/crossplane/crossplane/actions/runs/11484983049/job/31964036101?pr=6025) is failing with a bunch of:
```
Error: The copy of the 'for' variable "i" can be deleted (Go 1.22+) (copyloopvar)
```
Perhaps I should just keep the go version on `1.21`, instead of the bump to `1.22`, similar to how I did it for `release-1.15` in https://github.com/crossplane/crossplane/pull/6026.